### DISCUSSION TITLE
固定的图片水印

### DISCUSSION THREAD

**Discussion by Geo123abc** [wmjordan/PDFPatcher#106] [交流]
如附件所示，每一页都有一个相同的图片水印，分析发现是叠加的一个图片，这种水印可以使用PDF补丁丁解决吗？
https://wws.lanzouw.com/iTqtB05pdtpi
密码:5ygy

谢谢

**Comment by wmjordan** [comment]
这文档是很好处理的。

## 分析

1. 用文档结构探查器打开文件。
2. 展开 Pages，查看页面内的“命令”。
   <img width="281" alt="image" src="https://user-images.githubusercontent.com/12182730/171306500-5644fd7f-abae-4637-8233-b8444f00e97d.png">
1. 在“命令”内可以看到有两个“绘图状态”。
2. 两个“绘制对象”分别将对象“/Im642”和“/fzImg0”绘制到页面。
3. 展开“XObject”，可以看到上述两个对象。
4. 选中“XObject”内的“/fzImg0”，点击工具栏上的“查看”按钮，可以看到那就是你要消除的水印图片；“Im642”则是内容图片。
1. 再观察其它页面，结构是类似的，只是绘制对象中引用的图片不一样。
2. 因此，我们只需要去掉每页包含最后一幅图片的“绘图状态”就可以了。

## 处理

1. 用编辑器打开页面。
3. 点击“保存”按钮。
4. 设置清除页面结尾 1 条指令。
5. 再保存文档，就看到水印图片消失了。
